EventVisualisation/DataConverter/src/VisualisationEventROOTSerializer.cxx

```diff
@@ -0,0 +1,329 @@
+  long cluster_xyz;
+  int cluster_source;
+  float cluster_time;
+  clusters.Branch("source", &cluster_source);
+  clusters.Branch("time", &cluster_time);
+  clusters.Branch("xyz", &cluster_xyz);
+  for (auto cluster : event.getClustersSpan()) {
+    cluster_source = cluster.getSource();
+    cluster_time = cluster.Time();
+    cluster_xyz = xyzPos;
+    xyz.Fill(cluster.X(), cluster.Y(), cluster.Z());
+    xyzPos++;
+    clusters.Fill();
+  }
+
+  // Tracks
+  long track_xyz;     // first track point
+  int track_points;   // number of track points
+  int track_clusters; // number of track clusters
+  int track_source;
+  std::string track_GID;
+  float track_time;
+  int track_charge;
+  float track_theta;
+  float track_phi;
+  float track_eta;
+  int track_PID;
+
+  TTree tracks("tracks", "Tracks");
+  tracks.Branch("xyz", &track_xyz);
+  tracks.Branch("time", &track_time);
+  tracks.Branch("charge", &track_charge);
+  tracks.Branch("theta", &track_theta);
+  tracks.Branch("phi", &track_phi);
+  tracks.Branch("eta", &track_eta);
+  tracks.Branch("PID", &track_PID);
+  tracks.Branch("GID", &track_GID);
+  tracks.Branch("source", &track_source);
+  tracks.Branch("points", &track_points);
+  tracks.Branch("clusters", &track_clusters);
+
+  for (auto track : event.getTracksSpan()) {
+    track_xyz = xyzPos;
+    track_time = std::isnan(track.mTime) ? 0 : track.mTime;
+    track_charge = track.mCharge;
+    track_theta = std::isnan(track.mTheta) ? 0 : track.mTheta;
+    track_phi = std::isnan(track.mPhi) ? 0 : track.mPhi;
+    track_eta = std::isnan(track.mEta) ? 0 : track.mEta;
+    track_PID = track.mPID;
+    track_GID = track.mGID;
+    track_source = track.mSource;
+
+    xyz.Fill(track.mStartCoordinates[0], track.mStartCoordinates[1], track.mStartCoordinates[2]);
+    xyzPos++;
+    track_points = track.getPointCount();
+
+    for (size_t i = 0; i < track.getPointCount(); i++) {
+      xyz.Fill(track.mPolyX[i], track.mPolyY[i], track.mPolyZ[i]);
+      xyzPos++;
+    }
+    track_clusters = track.getClusterCount();
+    for (auto cluster : track.getClustersSpan()) {
+      xyz.Fill(cluster.X(), cluster.Y(), cluster.Z());
+      xyzPos++;
+    }
+    tracks.Fill();
+  }
+
+  // calorimeters
+  TTree calo("calo", "Calorimeters");
+  int calo_source;
+  float calo_time;
+  float calo_energy;
+  float calo_eta;
+  float calo_phi;
+  std::string calo_GID;
+  int calo_PID;
+
+  calo.Branch("source", &calo_source);
+  calo.Branch("time", &calo_time);
+  calo.Branch("energy", &calo_energy);
+  calo.Branch("eta", &calo_eta);
+  calo.Branch("phi", &calo_phi);
+  calo.Branch("GID", &calo_GID);
+  calo.Branch("PID", &calo_PID);
+
+  for (auto calorimeter : event.getCalorimetersSpan()) {
+    calo_source = calorimeter.getSource();
+    calo_time = calorimeter.getTime();
+    calo_energy = calorimeter.getEnergy();
+    calo_eta = calorimeter.getEta();
+    calo_phi = calorimeter.getPhi();
+    calo_GID = calorimeter.getGIDAsString();
```

Comment:
what is the reason of using string for GID, which is a simple integer?

Reply:
It was preserved from the previous version to keep interoperability with previous *.json files. I will address that in next (separate to that) PR when I will be able to test this change only. Please merge as it is and I will fix it soon.